### PR TITLE
Update Adform adapter to Prebid v1.0

### PR DESCRIPTION
**Pull request by Pupis**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Other

## Description of change
Update Adform adapter to Prebid v1.0

- contact email of the adapter’s maintainer
Scope.FL.Scripts@adform.com
- [x] official adapter submission

### CODE REVIEW

File: modules/adformBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      var key;
+      var url = [];
+
+      for (key in reqData) {
+        if (reqData.hasOwnProperty(key) && reqData[key]) { url.push(key, '=', reqData[key], '&'); }
+      }
+      return encode64(url.join('').slice(0, -1));
+    }
+
+    function encode64(input) {
```
Comment (snapwich) [inline modules/adformBidAdapter.js]: is there a reason you are manually performing base64 encoding over using native `btoa`?

File: modules/adformBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    for (i = 0, l = validBidRequests.length; i < l; i++) {
+      bid = validBidRequests[i];
+      if (bid.params.priceType === 'gross') {
+        netRevenue = 'gross';
+      }
+      for (j = 0, k = globalParams.length; j < k; j++) {
+        _key = globalParams[j][0];
+        _value = bid[_key] || bid.params[_key];
+        if (_value) {
+          bid[_key] = bid.params[_key] = null;
```
[the remark this answers is not in the document]
Comment (Pupis) [reply modules/adformBidAdapter.js]: Hi, changes were made to avoid mutating original bid objects.

File: modules/adformBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+const BIDDER_CODE = 'adform';
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [],
+  isBidRequestValid: function (bid) {
+    return !!(bid.params.mid);
+  },
+  buildRequests: function (validBidRequests) {
+    var i, l, j, k, bid, _key, _value, reqParams;
```
Comment (mkendall07) [inline modules/adformBidAdapter.js]: nit: prefer `const` | `let`

File: modules/adformBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    var request = [];
+    var globalParams = [ [ 'adxDomain', 'adx.adform.net' ], [ 'fd', 1 ], [ 'url', null ], [ 'tid', null ] ];
+    var netRevenue = 'net';
+    var bids = JSON.parse(JSON.stringify(validBidRequests));
```
Comment (mkendall07) [inline modules/adformBidAdapter.js]: we have a copy function in utils `deepClone` that could be used. Although this is probably fine.

**Comment by rmloveland** [comment]
Added the needs-docs label since it's not showing up on http://prebid.org/download.html in the 1.2.0 dropdown due to lack of the page having the "prebid 1.0 support"  key in the page's YAML header.

Let me know if there is another better way to do this.

**Pull request merged**